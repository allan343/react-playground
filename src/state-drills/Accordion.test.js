import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Accordion from './Accordion';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';


describe(`Accordions Component`, () => {
    // array of objects each with a name and content
    const sections = [
        {
          title: 'Section 1',
          content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
        },
        {
          title: 'Section 2',
          content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
        },
        {
          title: 'Section 3',
          content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
        },
      ]
  
    /* smoke test not shown */
  
    it('renders the first tab by default', () => {
      const tree = renderer.create(<Accordion />).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders no sections as active by default', () => {
        const tree = renderer.create(<Accordion sections = {sections} />).toJSON()
        expect(tree).toMatchSnapshot()
      })

      it('opens any clicked tab', () => {
        const wrapper = shallow(<Accordion sections = {sections} />)
        wrapper.find('button').at(1).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
      })

      it('only opens the last section when multiple sections have been clicked', () => {
        const wrapper = shallow(<Accordion sections = {sections} />)
        wrapper.find('button').at(0).simulate('click')
        wrapper.find('button').at(1).simulate('click')
        wrapper.find('button').at(2).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
      })
  })