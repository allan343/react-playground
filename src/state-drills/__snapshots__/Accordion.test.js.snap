// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordions Component only opens the last section when multiple sections have been clicked 1`] = `
<ul>
  <li>
    <button
      key="0"
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li>
    <button
      key="1"
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li>
    <button
      key="2"
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
  <div
    className="content"
  >
    Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
  </div>
</ul>
`;

exports[`Accordions Component opens any clicked tab 1`] = `
<ul>
  <li>
    <button
      key="0"
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li>
    <button
      key="1"
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li>
    <button
      key="2"
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
  <div
    className="content"
  >
    Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
  </div>
</ul>
`;

exports[`Accordions Component renders no sections as active by default 1`] = `
<ul>
  <li>
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordions Component renders the first tab by default 1`] = `
<ul>
  0
</ul>
`;
